File: mo-system/mo-system-api/src/main/java/ink/ckx/mo/system/api/model/entity/SysMenu.java
package ink.ckx.mo.system.api.model.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import ink.ckx.mo.common.core.base.BaseEntity;
import ink.ckx.mo.common.web.enums.StatusEnum;
import ink.ckx.mo.system.api.enums.MenuTypeEnum;
import lombok.Data;

/**
 * 菜单实体类
 *
 * @author chenkaixin
 */
@Data
public class SysMenu extends BaseEntity {

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    private Long parentId;

    private String treePath;

    private String name;

    private MenuTypeEnum type;

    private String path;

    private String component;

    private String perm;

    private String icon;

    private Integer sort;

    private Integer alwaysShow;

    private StatusEnum visible;
}

File: mo-admin/mo-admin-api/src/main/kotlin/ink/ckx/mo/admin/api/model/entity/SysMenu.kt
package ink.ckx.mo.admin.api.model.entity

import com.baomidou.mybatisplus.annotation.IdType
import com.baomidou.mybatisplus.annotation.TableId
import ink.ckx.mo.admin.api.enums.MenuTypeEnum
import ink.ckx.mo.common.core.base.BaseEntity
import ink.ckx.mo.common.web.enums.StatusEnum

/**
 * 菜单实体类
 *
 * @author chenkaixin
 */
data class SysMenu(
    @TableId(type = IdType.ASSIGN_ID)
    var id: Long? = null,
    var parentId: Long? = null,
    var treePath: String? = null,
    var name: String? = null,
    var type: MenuTypeEnum? = null,
    var path: String? = null,
    var component: String? = null,
    var perm: String? = null,
    var icon: String? = null,
    var sort: Int? = null,
    var alwaysShow: Int? = null,
    var visible: StatusEnum? = null,
) : BaseEntity()